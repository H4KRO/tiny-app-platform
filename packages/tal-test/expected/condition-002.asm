main()
  entry:
    Literal 1
    DeclareLocal      name: a, mutable: true, hasInitialValue: true
    Pop               inBlock: false
    Literal 1
    DeclareLocal      name: b, mutable: true, hasInitialValue: true
    Pop               inBlock: false
    Literal false
    JumpTrue          label: if_true_1
    Jump              label: if_false_2
  if_true_1:
    ScopeEnter
    Literal 2
    SetLocal          name: a
    ScopeLeave        inBlock: false, count: 1
    Jump              label: if_end_3
  if_false_2:
    Literal null
    Jump              label: if_end_3
  if_end_3:
    Pop               inBlock: false
    Local             name: a
    Pop               inBlock: false
    Local             name: b
