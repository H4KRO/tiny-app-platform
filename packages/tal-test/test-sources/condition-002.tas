var a = 1
var b = 1
if (false) {
    set a = 2
}
a
b
