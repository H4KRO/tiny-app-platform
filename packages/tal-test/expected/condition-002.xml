<program>
  <declarelocal>
    <literal />
  </declarelocal>
  <declarelocal>
    <literal />
  </declarelocal>
  <if>
    <literal />
    <block>
      <assign>
        <local />
        <literal />
      </assign>
    </block>
  </if>
  <local />
  <local />
</program>
